rdkit-stubs/Chem/BuildFragmentCatalog.pyi

```diff
@@ -1,98 +1,204 @@
+    acts: Optional[Sequence[int]] = None,
+    nActs: int = 2,
+    reportFreq: int = 10,
+    biasList = None,
+    collectFps: int = 0,
+): 
+    """ 
+    calculates info gains by constructing fingerprints
+    *DOC*
+
+    Returns a 2-tuple:
+        1) gains matrix
+        2) list of fingerprints
+    """
 def CalcGainsFromFps(
     suppl,
     fps,
-    topN: int = ...,
-    actName: str = ...,
-    acts: Incomplete | None = ...,
-    nActs: int = ...,
-    reportFreq: int = ...,
-    biasList: Incomplete | None = ...,
-): ...
-def OutputGainsData(outF, gains, cat, nActs: int = ...) -> None: ...
-def ProcessGainsData(inF, delim: str = ..., idCol: int = ..., gainCol: int = ...): ...
+    topN: int = -1,
+    actName: str = '',
+    acts: Optional[Sequence[int]] = None,
+    nActs: int = 2,
+    reportFreq: int = 10,
+    biasList = None,
+):
+    """ 
+    calculates info gains from a set of fingerprints
+
+    *DOC*
+    """
+def OutputGainsData(outF: IOBase, gains, cat, nActs: int = 2) -> None: ...
+def ProcessGainsData(inF: IOBase, delim: str = ',', idCol: int = 0, gainCol: int = 1):
+    """reads a list of ids and info gains out of an input file"""
 def ShowDetails(
     catalog,
     gains,
-    nToDo: int = ...,
-    outF=...,
-    idCol: int = ...,
-    gainCol: int = ...,
-    outDelim: str = ...,
-) -> None: ...
+    nToDo: int = -1,
+    outF: IOBase = sys.stdout,
+    idCol: int = 0,
+    gainCol: int = 1,
+    outDelim: str = ',',
+) -> None:
+    """
+    gains should be a sequence of sequences.  The idCol entry of each
+    sub-sequence should be a catalog ID.  _ProcessGainsData()_ provides
+    suitable input.
+    """
 def SupplierFromDetails(details): ...
 def Usage() -> None: ...
 
 class RunDetails:
-    numMols: int
-    doBuild: int
-    doSigs: int
-    doScore: int
-    doGains: int
-    doDetails: int
-    catalogName: Incomplete
-    onBitsName: Incomplete
-    scoresName: Incomplete
-    gainsName: Incomplete
-    dbName: str
-    tableName: Incomplete
-    detailsName: Incomplete
-    inFileName: Incomplete
-    fpName: Incomplete
-    minPath: int
-    maxPath: int
-    smiCol: int
-    actCol: int
-    nameCol: int
-    hasTitle: int
-    nActs: int
-    nBits: int
-    delim: str
-    biasList: Incomplete
-    topN: int
-
-def ParseArgs(details) -> None: ...
+  numMols: int = -1
+  doBuild: int = 0
+  doSigs: int = 0
+  doScore: int = 0
+  doGains: int = 0
+  doDetails: int = 0
+  catalogName: Optional[str] = None
+  onBitsName: Optional[str] = None
+  scoresName: Optional[str] = None
+  gainsName: Optional[str] = None
+  dbName: str = ''
+  tableName: Optional[str] = None
+  detailsName: Optional[str] = None
+  inFileName: Optional[str] = None
+  fpName: Optional[str] = None
+  minPath: int = 2
+  maxPath: int = 6
+  smiCol: int = 1
+  actCol: int = -1
+  nameCol: int = -1
+  hasTitle: int = 1
+  nActs: int = 2
+  nBits: int = -1
+  delim: str = ','
+  biasList: Optional[tuple] = None
+  topN: int = -1
```

Comment:
```suggestion
    numMols: int = -1
    doBuild: int = 0
    doSigs: int = 0
    doScore: int = 0
    doGains: int = 0
    doDetails: int = 0
    catalogName: Optional[str] = None
    onBitsName: Optional[str] = None
    scoresName: Optional[str] = None
    gainsName: Optional[str] = None
    dbName: str = ''
    tableName: Optional[str] = None
    detailsName: Optional[str] = None
    inFileName: Optional[str] = None
    fpName: Optional[str] = None
    minPath: int = 2
    maxPath: int = 6
    smiCol: int = 1
    actCol: int = -1
    nameCol: int = -1
    hasTitle: int = 1
    nActs: int = 2
    nBits: int = -1
    delim: str = ','
    biasList: Optional[tuple] = None
    topN: int = -1
```

Reply:
Consider running `black .` on in the root of the repo to automatically keep formatting consistent.

Reply:
I update the format with `black .` as you suggested.